R/tm_outliers.R

```diff
@@ -456,6 +458,29 @@ srv_outliers <- function(id, data, reporter, filter_panel_api, outlier_var,
           remove_pipe_null()
       )
 
+      # ANL_OUTLIER_EXTENDED is the base table
+      qenv <- teal.code::eval_code(
+        qenv,
+        substitute(
+          expr = {
+            ANL_OUTLIER_EXTENDED <- dplyr::left_join( # nolint object_name_linter
+              ANL_OUTLIER,
+              dplyr::select(
+                dataname,
+                dplyr::setdiff(names(dataname), dplyr::setdiff(names(ANL_OUTLIER), join_keys))
+              ),
+              by = join_keys
+            )
+          },
+          env = list(
+            dataname = as.name(dataname_first),
+            join_keys = as.character(get_join_keys(data)$get(dataname_first)[[dataname_first]])
+          )
+        )
+      ) %>%
+        # used to display table when running show-r-code code
+        teal.code::eval_code(quote(ANL_OUTLIER_EXTENDED[ANL_OUTLIER_EXTENDED$is_outlier_selected, ]))
```

Comment:
`ANL_OUTLIER_EXTENDED[ANL_OUTLIER_EXTENDED$is_outlier_selected, ]`Show all columns in the dataset, including those not displayed on the dashboard. 

I believe that the code should print a table that displays on the dashboard, as well as any user-selected columns.

Reply:
@kartikeyakirar this may have a UX side effect as any change to the columns being selected will clear the selected data points.

My suggestion is to add the outlier table print after the plot call, otherwise, any column change will trigger reactivity in the `<plot>_q` and re-render the plot.

The downside is that this approach requires duplicate code only _(column selection)_

![image](https://github.com/insightsengineering/teal.modules.general/assets/211358/9765b98c-d4bd-42d4-869b-dd3ad7025eb4)

WDYT?

Reply:
> Show all columns in the dataset, including those not displayed on the dashboard.

this problem is fixed. I can see table displayed on dashboard.

Reply:
> any change to the columns being selected will clear the selected data points.

I believe there's no necessity for us to proceed with this step, as we're unable print  user selections within the ggplot. Thus, there's no need to gather the selected datapoints in reproducible code. In my opinion, doing so might lead to confusion when working with reproducible code. I'd appreciate your input on this, @cicdguy, particularly with regard to displaying datapoints from outlier selections in the plot.

_Image to capture outlier points on table_
![image](https://github.com/insightsengineering/teal.modules.general/assets/6700955/019e5ed7-bdef-480f-89cb-e5d2cdb358c4)

Reply:
@kartikeyakirar 

> any change to the columns being selected will clear the selected data points.

My statement wasn't super clear. This relates to the placement of the table-print code in `qenv`, as if we place the code for `table-print` before the `plot-print`, then it will add reactivity that re-renders the plot every time a column is selected/deselected _(which causes all brushed selection being lost)_

Hence, the `table-print` is located after the `plot-print`.